bugwarrior/services/youtrack.py

```diff
@@ -0,0 +1,188 @@
+from __future__ import absolute_import
+import re
+import six
+
+import requests
+from jinja2 import Template
+
+from bugwarrior.config import asbool, die
+from bugwarrior.services import IssueService, Issue, ServiceClient
+
+import logging
+
+log = logging.getLogger(__name__)
+
+
+class YoutrackIssue(Issue):
+    ISSUE = 'youtrackissue'
+    SUMMARY = 'youtracksummary'
+    URL = 'youtrackurl'
+    PROJECT = 'youtrackproject'
+    NUMBER = 'youtracknumber'
+
+    UDAS = {
+        ISSUE: {
+            'type': 'string',
+            'label': 'YouTrack Issue'
+        },
+        SUMMARY: {
+            'type': 'string',
+            'label': 'YouTrack Summary',
+        },
+        URL: {
+            'type': 'string',
+            'label': 'YouTrack URL',
+        },
+        PROJECT: {
+            'type': 'string',
+            'label': 'YouTrack Project'
+        },
+        NUMBER: {
+            'type': 'string',
+            'label': 'YouTrack Project Issue Number'
+        },
+    }
+    UNIQUE_KEY = (URL,)
+
+    def _get_record_field(self, field_name):
+        for field in self.record['field']:
+            if field['name'] == field_name:
+                return field
+
+    def _get_record_field_value(self, field_name, field_value='value'):
+        field = self._get_record_field(field_name)
+        if field:
+            return field[field_value]
```

Comment:
Admittedly the function is redundant for the set of features currently implemented, but if you'll take a look at the sample output from a YouTrack REST endpoint
https://gist.github.com/babelop/d4afacfb0c753c2fc3cd6b7f702e0b78

Complex field types include extra meta-data like valueId and color stored at the same hierarchal level as value, and would require multiple trips through the array if using the "value" convenience method alone. The ability to return the whole record block is ideal for future use in consuming, for example, priority data.

However, I defer to your preference if you'd like to see it pruned at this stage.